django/common/adapters/fhir_api.py

```diff
@@ -52,6 +63,30 @@ def create(self, resource_type: str, payload: dict, auth_token=None):
         response.raise_for_status()
         return response.json()
 
+    def validate(self, resource_type: str, payload: dict, auth_token=None):
+        """Calls the /<resource_type>/$validate endpoint of HAPI FHIR.
+        Note that this function does not raise an exception if the status is not 2XX.
+
+        Args:
+            resource_type (str): the resource type
+            payload (dict): the FHIR instance
+            auth_token ([type], optional): The authentication token to access FHIR API.
+            Defaults to None.
+
+        Returns:
+            (dict): OperationOutcome containing the details about validation errors.
+        """
+        headers = {**self._headers, "Authorization": f"Bearer {auth_token}"} if auth_token else self._headers
```

Comment:
nit: Maybe we could factorize shared code into a `query` method?

Reply:
done